jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -241,7 +241,14 @@ MethodHandle cloneWithNewType(MethodType newType) {
 	/*[IF ]*/
 	// Until the JIT can synthesize calls to virtual methods, we must synthesize calls to these static ones instead
 	/*[ENDIF]*/
-	private static MethodHandle asType(MethodHandle mh, MethodType newType) { return mh.asType(newType); }
+	private static MethodHandle asType(MethodHandle mh, MethodType newType) {
+		/* JIT can easily propagate type information and fold the if when it can prove early return always happen */
```

Comment:
Can you expand on this comment to indicate that the early check saves the JIT from having to inline full `asType` call?